crypto/src/keys/diversifier.rs

```diff
@@ -177,6 +178,18 @@ impl From<u32> for AddressIndex {
     }
 }
 
+impl FromStr for AddressIndex {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> anyhow::Result<Self> {
+        let i: u32 = s.parse()?;
```

Comment:
Is this meant not to support string representations of ephemeral addresses? If so, should we add a comment here to that effect? Otherwise LGTM 😎

Reply:
That was not my intention, simply an oversight. It's not clear to me what changes would be necessary for ephemeral address, so I'll tack on a comment for now!